animal.go

```diff
@@ -43,9 +45,28 @@ func GetUserYesOrNo(question string, r io.Reader) (string, error) {
 
 func NextQuestion(q string, r string) string {
 
+	Data := map[string]Question{
```

Comment:
It'll be a little more efficient to define this variable at package level, so that it's not created and destroyed every time the NextQuestion function is called.

Reply:
@bitfield thanks for the feedback - had struggled initially doing that but now able to do it. Pushing a new commit.